### PR TITLE
Add a tutorial for OAuth2

### PR DESCRIPTION
**Pull request by ekoleda-codaio**
Staged here: https://coda-packs-sdk--2062.com.readthedocs.build/en/2062/tutorials/build/oauth/

Make sure to review [oauth.md](https://github.com/coda/packs-sdk/pull/2062/files#diff-4b5340a1702722e93254670138839a9076051f05c295e6c8ab74cb7e9fb27cd4), since the diff isn't loaded by default.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -160,7 +160,7 @@ docs: typedoc generated-documentation build-mkdocs
 
 .PHONY: view-docs
 view-docs:
-	PYTHONPATH=${ROOTDIR} PIPENV_IGNORE_VIRTUALENVS=1 MK_DOCS_SITE_URL=http://localhost:8000/packs-sdk expect -c 'spawn pipenv run mkdocs serve; expect "Serving on"; exec open "http://localhost:8000"; interact'
+	PYTHONPATH=${ROOTDIR} PIPENV_IGNORE_VIRTUALENVS=1 MK_DOCS_SITE_URL=http://localhost:8000/packs-sdk expect -c 'set timeout 60; spawn pipenv run mkdocs serve; expect "Serving on"; exec open "http://localhost:8000"; interact'
```
Comment (jonathan-codaio) [inline Makefile:163]: Is this timeout intentional?

File: docs/tutorials/build/oauth.md [hunk 2]
```diff
@@ -0,0 +1,573 @@
...
+
+    </div>
+    </section>
+
+=== ":material-numeric-2-circle: Token URL"
+
+    <section class="tutorial-row" markdown>
+    <div markdown>
+
+    After the user approves the access request, the Pack is given a temporary code that can be exchanged for an access token. To perform that exhcange the Pack makes a request to the API's token URL.
```
Comment (jonathan-codaio) [inline docs/tutorials/build/oauth.md:78]: Typo, exchange

File: docs/tutorials/build/oauth.md [hunk 3]
```diff
@@ -0,0 +1,573 @@
...
+    ```
+    https://coda.io/packsAuth/oauth2
+    ```
+
+    Click the **Save settings** button to save your changes.
+
+    </div>
+    <div markdown>
+
+    <img src="../../../images/tutorial_oauth_settings.png" srcset="../../../images/tutorial_oauth_settings_2x.png 2x" class="screenshot" alt="The settings for the new application">
```
Comment (jonathan-codaio) [inline docs/tutorials/build/oauth.md:200]: I assume the client id and secret in the screenshot have been reset so that they're no longer usable?

**Review by jonathan-codaio — APPROVED** [review]
Nice

File: docs/tutorials/build/oauth.md (hunk 3 above)
Comment (ekoleda-codaio) [reply docs/tutorials/build/oauth.md:200]: It is now! Was planning to, but hadn't remembered until this reminder.

**Pull request merged**